plugins/com.google.cloud.tools.eclipse.appengine.localserver/src/com/google/cloud/tools/eclipse/appengine/localserver/server/LocalAppEngineServerLaunchConfigurationDelegate.java

```diff
@@ -44,7 +56,63 @@ public void launch(ILaunchConfiguration configuration, String mode, ILaunch laun
     TargetPlatform.showConsole(console);
     console.clearConsole();
 
+    LocalAppEngineServerDelegate serverDelegate = LocalAppEngineServerDelegate.getAppEngineServer(server);
+    int debugPort = -1;
+    if (ILaunchManager.DEBUG_MODE.equals(mode)) {
+      debugPort = getDebugPort();
+
+      try {
+        runDebugTarget(serverDelegate, modules[0].getProject().getName(), debugPort);
+      } catch (CoreException e) {
+        Activator.logError(e);
+      }
+    }
+
     // Start server
-    serverBehaviour.startDevServer(runnables, console.newMessageStream());
+    serverBehaviour.startDevServer(runnables, console.newMessageStream(), mode, debugPort);
+  }
+
+  private void runDebugTarget(LocalAppEngineServerDelegate serverDelegate, String projectName, int port)
+      throws CoreException {
+    ILaunchConfigurationWorkingCopy remoteDebugLaunchConfig = createRemoteDebugLaunchConfiguration(serverDelegate,
+        projectName,
+        Integer.toString(port));
+    remoteDebugLaunchConfig.launch(ILaunchManager.DEBUG_MODE, null);
+  }
+
+  private ILaunchConfigurationWorkingCopy createRemoteDebugLaunchConfiguration(LocalAppEngineServerDelegate serverDelegate,
+      final String projectName, final String port) throws CoreException {
```

Comment:
why are these arguments final? Google style is to omit final on arguments and local variables unless it's required (e.g. by use in an inner class)